### DISCUSSION TITLE
`theme.locale` or `theme.language`?

### DISCUSSION THREAD

**Discussion by Guts** [squidfunk/mkdocs-material#6453] [Q&A] [answered]
Hello,

I'm trying to solve an issue on the RSS Plugin (see https://github.com/Guts/mkdocs-rss-plugin/pull/212, https://github.com/Guts/mkdocs-rss-plugin/pull/206 and https://github.com/Guts/mkdocs-rss-plugin/issues/205) related to the site language.

I'm a bit confused on the Mkdocs setting to use:

- the Mkdocs `theme.locale`: https://www.mkdocs.org/user-guide/configuration/#locale
- the Material for Mkdocs `theme.language`: https://squidfunk.github.io/mkdocs-material/setup/changing-the-language/#site-language

My question is mainly the following: what's the rationale behind using a custom `language` option instead of using the upstream Mkdocs `locale`? is the `language` setting defining the `locale`? I need to understand before moving on.

cc @alexvoss

**Comment by alexvoss** [comment]
That is a good question. If the answer turns out to be "this is historical" then perhaps aligning with MkDocs would be good and we could get Material [mentioned as one of the themes that supports i18n](https://www.mkdocs.org/user-guide/localizing-your-theme/#supported-locales)? Just my 2p.

**Reply by squidfunk** [reply]
Answered in https://github.com/squidfunk/mkdocs-material/issues/4253#issuecomment-1215177476

**Reply by squidfunk** [reply, marked as answer]
To expand on this: we might switch to `locale`, maybe in one of the next major releases. However, we should also re-evaluate whether we can somehow use a better translation mechanism that we currently have, but it is definitely not a priority, since it works reasonably well and makes contributing translations ridiculously simple. This is also why we're already at 65 translations.

If somebody wants to work on this, we can consider a PR. Note that we have translations in the frontend (mainly search).

**Reply by Guts** [reply]
Thank you for your quick answer and the explanation.

I've pushed a custom behavior for Material theme: https://github.com/Guts/mkdocs-rss-plugin/blob/560d7067527a4ca0bd96fad94f41fc0c3876ae5c/mkdocs_rss_plugin/util.py#L626-L637
